From 4b7651e9e43edfe11e813d39ace15a574428f93d Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Sun, 27 Jun 2021 16:16:00 +0300
Subject: [PATCH] [SW-0] sec: fix function signature

One implementation of the "preboot_security_dynamic_img_spsram_open"
function was not matched the function signature. Therefore the weak
function was used instead. The security.h header was not included as
well, so there was no compilation warning as well.

Change-Id: I30468dec8a530993833c50e88f0a3128f142364b
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/arch/gaudi/security.c b/arch/gaudi/security.c
index e572749..b67dfb8 100644
--- a/arch/gaudi/security.c
+++ b/arch/gaudi/security.c
@@ -63,6 +63,7 @@
 #include <hl_fw_if.h>
 #include <preboot/fit.h>
 #include <memory.h>
+#include <security.h>
 
 #include "common.h"
 
@@ -1767,7 +1768,7 @@
 	preboot_secure_pcie_rr(cfg->rr, cfg->min, cfg->max);
 }
 
-void preboot_security_dynamic_img_spsram_open(enum hl_upd_cmd cmd)
+void preboot_security_dynamic_img_spsram_open(void)
 {
 	preboot_secure_pcie_rr(preboot_sec_dyn_cfg[SPSRAM_IMAGE].rr, 0, 0);
 }
